### ISSUE TITLE
`use_su_login` not working as expected?

### ISSUE THREAD

**Issue by Fizzadar** [Fizzadar/pyinfra#426]
**Describe the bug**
See: https://github.com/Fizzadar/pyinfra/issues/412#issuecomment-685092335

**Comment by Fizzadar** [comment]
This is potentially expected behaviour - at least on MacOS I can't get `su -l` to execut as expected when passing a command:

```sh
root$ su -l nick
nick$ echo $PYENV_ROOT
/Users/nick/.pyenv
```

vs

```sh
root$ su -l nick -c 'echo$PYENV_ROOT'
     # nothing output
```

**Comment by Fizzadar** [comment]
Interestingly this works as expected on freebsd:

```sh
vagrant ssh freebsd12
Last login: Thu Sep  3 15:19:51 2020 from 10.0.2.2
hi
[vagrant@freebsd12 ~]$ su -l ^C
[vagrant@freebsd12 ~]$ sudo su
[root@freebsd12 /usr/home/vagrant]# whoami
root
[root@freebsd12 /usr/home/vagrant]# su -l vagrant
hi
[vagrant@freebsd12 ~]$ logout
[root@freebsd12 /usr/home/vagrant]# su -l vagrant -c ''
hi
[root@freebsd12 /usr/home/vagrant]# su -l vagrant -c 'echo hi'
hi
hi
```

And with pyinfra:

```sh
$ pyinfra @vagrant/freebsd12 server.shell 'echo hello' sudo=true su_user=vagrant -vvv
...
--> Starting operation: Server/Shell ('echo hello',)
[@vagrant/freebsd12] >>> sudo -H -n su vagrant -c 'sh -c '"'"'echo hello'"'"''
[@vagrant/freebsd12] hello
    [@vagrant/freebsd12] Success

# with su login enabled

$ pyinfra @vagrant/freebsd12 server.shell 'echo hello' sudo=true su_user=vagrant use_su_login=true -vvv
...
--> Starting operation: Server/Shell ('echo hello',)
[@vagrant/freebsd12] >>> sudo -H -n su -l vagrant -c 'sh -c '"'"'echo hello'"'"''
[@vagrant/freebsd12] hi
[@vagrant/freebsd12] hello
    [@vagrant/freebsd12] Success
```

So I can only conclude that the MacOS variant of `su` does not respect the `-l` flag when executing a command directly. Annoying gotcha perhaps?

**Comment by Fizzadar** [comment]
Based on the above - I no longer think this is a bug, but a gotcha with various systems. I think the best next step here is to add a warning when using `use_su_login` highlighting the potential issues.

**Comment by Fizzadar** [comment]
Warning added in https://github.com/Fizzadar/pyinfra/commit/bb60661046e552c993857432c0bc8db5e58f1924 which will be released imminently in `v1.1.2`.
